pytools/tag.py

```diff
@@ -127,6 +128,92 @@ class UniqueTag(Tag):
 
 
 TagsType = FrozenSet[Tag]
+TagOrIterableType = Union[Iterable[Tag], Tag, None]
+T_co = TypeVar("T_co", bound="Taggable")
+
+
+class Taggable():
+    """
+    Parent class for objects with a `tags` attribute.
+
+    .. attribute:: tags
+
+        A :class:`frozenset` of :class:`Tag` instances
+
+    .. method:: copy
+    .. method:: tagged
+    .. method:: without_tags
+
+    .. versionadded:: 2020.4.4
+    """
+
+    def __init__(self, tags: TagOrIterableType = None):
+        if isinstance(tags, Tag):
+            self.tags = frozenset([tags])
+        elif tags is None:
+            self.tags = frozenset()
+        else:
+            self.tags = frozenset(tags)
+        self._check_uniqueness()
+
+    def _check_uniqueness(self):
+        unique_tags = {tag for tag in self.tags if isinstance(tag, UniqueTag)}
+        for tag1 in unique_tags:
+            for tag2 in unique_tags:
```

Comment:
That's O(n^2). Use set intersection instead.

Reply:
I'm not seeing how this works with inheritance. Say I have `UniqueTagChild` and `UniqueTagChildChild`. Then  `{UniqueTagChild}.intersection({UniqueTagChildChild})` gives `set()` rather than `{UniqueTagChild}`.

Reply:
That's a fair point. Could you add a comment explaining that?